apps/modernization-ui/src/SkipLink/SkipLinkContext.tsx

```diff
@@ -24,14 +26,30 @@ interface SkipLinkProviderProps {
 
 export const SkipLinkProvider = ({ children }: SkipLinkProviderProps) => {
     const [currentFocusTarget, setCurrentFocusTarget] = useState('');
+    const [previousFocusTarget, setPreviousFocusTarget] = useState('');
 
     const skipTo = (id: string) => {
         setCurrentFocusTarget(id);
     };
 
+    const previousTarget = (id: string) => {
+        setPreviousFocusTarget(id);
+    };
+
+    useEffect(() => {
+        const handleKeyDown = (event: KeyboardEvent) => {
+            if (event.altKey && event.keyCode === 88) {
+                event.preventDefault();
+                focusedTarget(previousFocusTarget);
+            }
+        };
+        window.addEventListener('keydown', handleKeyDown);
+    }, [previousFocusTarget]);
```

Comment:
issue (blocker):  This is going to add a _new_ event listener each time the `previousFocusTarget` is changed.  A clean up function should be returned whenever a hook is interacting with an external system, see [useEffect](https://react.dev/reference/react/useEffect)